### PR TITLE
chore(localdebug): show task error in output and notification

### PR DESCRIPTION
**Pull request by XiaofuHuang**
show task error in output and notification
image
image

Updated: 
image

### CODE REVIEW

**Review by swatDong — DISMISSED** [review]
Code change looks good but questions on UI:
- Is the task terminal shown by default? Or it's `reveal: never`?
- Any quick way to avoid showing `[output panel](command:...)`?

**Comment by XiaofuHuang** [comment]
> Code change looks good but questions on UI:
> 
> * Is the task terminal shown by default? Or it's `reveal: never`?
> * Any quick way to avoid showing `[output panel](command:...)`?

When the task contains `reveal: never`, the terminal will not be shown, but user can see it by manually opening this terminal.

**Comment by swatDong** [comment]
> > Code change looks good but questions on UI:
> > 
> > * Is the task terminal shown by default? Or it's `reveal: never`?
> > * Any quick way to avoid showing `[output panel](command:...)`?
> 
> When the task contains `reveal: never`, the terminal will not be shown, but user can see it by manually opening this terminal.

Just to ensure the default value is `never` so this error message is only shown on few cases. Keep consistency of all non-background debug tasks that output panel is the main window.

**Review by kuojianlu — APPROVED** [review]

**Comment by XiaofuHuang** [comment]
> > > Code change looks good but questions on UI:
> > > 
> > > * Is the task terminal shown by default? Or it's `reveal: never`?
> > > * Any quick way to avoid showing `[output panel](command:...)`?
> > 
> > 
> > When the task contains `reveal: never`, the terminal will not be shown, but user can see it by manually opening this terminal.
> 
> Just to ensure the default value is `never` so this error message is only shown on few cases. Keep consistency of all non-background debug tasks that output panel is the main window.

Yes, `reveal` is not `never`  only in ngrok task .

**Review by swatDong — APPROVED** [review]

File: packages/vscode-extension/src/debug/constants.ts [hunk 1]
```diff
@@ -89,3 +89,5 @@ export class LaunchUrl {
 export const teamsAppIdPlaceholder = "${teamsAppId}";
 export const teamsAppInternalIdPlaceholder = "${teamsAppInternalId}";
 export const accountHintPlaceholder = "${account-hint}";
+
+export const outputPanelLink = "[output panel](command:fx-extension.showOutputChannel)";
```
Comment (tecton) [inline packages/vscode-extension/src/debug/constants.ts:93]: Does `[output panel]` need to be localized?

**Comment by kuojianlu** [comment]
I am thinking do we need to show terminal when the task fails? Seems output channel is great enough.

**Comment by XiaofuHuang** [comment]
> I am thinking do we ready need to show error messages in terminal when the task fails? Seems output channel is great enough.

If the user manually set  `reveal` / manually open the terminal, the user may be confused about why the task is failed. The message tell the user go to output panel to see the detail.

**Review by kuojianlu — APPROVED** [review]

**Review by swatDong — APPROVED** [review]

**Review by kuojianlu — APPROVED** [review]

File: packages/vscode-extension/src/debug/constants.ts (hunk 1 above)
Comment (XiaofuHuang) [reply packages/vscode-extension/src/debug/constants.ts]: Updated to localize output panel

**Review by tecton — APPROVED** [review]

**Pull request merged**